packages/collector/Collector.php

```diff
@@ -0,0 +1,250 @@
+<?php
+/**
+ * @package Collector
+ * @version 0.0.0
+ */
+/*
+Plugin Name: Collector
+Plugin URI: https://github.com/seanmorris/collector
+Description: Packages your WordPress install and sends it to Playground.
+Author: Sean Morris
+Version: 0.0.0
+Author URI: https://github.com/seanmorris/
+*/
+const COLLECTOR_DOWNLOAD_PATH = '?page=collector_download_package';
+const COLLECTOR_ADMIN_PAGE_SLUG = 'collector_render_playground_page';
+const COLLECTOR_PLAYGROUND_PACKAGE = 'https://playground.wordpress.net/client/index.js';
+const TRANSLATE_DOMAIN = 'playground-collector';
+const ADMIN_PAGE_CAPABILITY = 'manage_options';
+
+global $wp_version;
+
+define('COLLECTOR_WP_VERSION', $wp_version);
+define('COLLECTOR_PHP_VERSION', implode('.',sscanf(phpversion(), '%d.%d')));
+
+require __DIR__ . '/vendor/autoload.php';
+require __DIR__ . '/Collector_Content.php';
+require __DIR__ . '/Collector_Db.php';
+require __DIR__ . '/Collector_Helpers.php';
+require __DIR__ . '/Collector_Zip.php';
+
+add_action('admin_menu', 'collector_plugin_menu');
+add_action('plugins_loaded', 'collector_plugins_loaded');
+add_filter('plugin_install_action_links', 'collector_plugin_install_action_links', 10, 2);
+add_filter('plugins_api_args', 'collector_plugins_api_args', 10, 2);
+
+function get_collector_admin_page_url() {
+	return admin_url(COLLECTOR_DOWNLOAD_PATH);
+}
+
+function collector_plugins_loaded()
+{
+	if(!current_user_can(ADMIN_PAGE_CAPABILITY))
+	{
+		return;
+	}
+
+	if(home_url($_SERVER['REQUEST_URI']) === get_collector_admin_page_url())
```

Comment:
Is this how plugins typically route requests? Something seems weird about that but WordPress surprised me more than once.

Reply:
No, they typically use API endpoints or at least register admin pages.
I didn't want to modify this part to much in this PR, but we should address it in the future to avoid our custom security checks and just rely on WP.

Reply:
Turns out we can't use the rest API here because we need a zip URL. The rest API would require us to have a blueprint step where we can pass a file previously downloaded using `fetch`.

I updated the code to use the admin page to generate ZIPs which would be a more standard solution. 8adf17168f645d5751085adc8f9a19d32c68eba0